File: 1886-determine-whether-matrix-can-be-obtained-by-rotation/1886-determine-whether-matrix-can-be-obtained-by-rotation.java
class Solution {
    public boolean findRotation(int[][] mat, int[][] target) {
        int n= mat.length;
        for(int r=0; r<4; r++){
            if(Arrays.deepEquals(mat,target)){
                return true;
            }
        
        for(int i=0; i<n;i++){
            for(int j=i+1;j<n;j++){
                int temp = mat[i][j];
                mat[i][j]=mat[j][i];
                mat[j][i]=temp;
                
            }
        }
        for(int i=0;i<n;i++){
            int left = 0;
            int right = n-1;
            while(left<right){
                int temp = mat[i][left];
                mat[i][left]=mat[i][right];
                mat[i][right]=temp;
                left++;
                right--;
            }
        }
        }
        return false;
    }
}

File: 1886-determine-whether-matrix-can-be-obtained-by-rotation/1886-determine-whether-matrix-can-be-obtained-by-rotation.py
class Solution(object):
    def findRotation(self, mat, target):
        """
        :type mat: List[List[int]]
        :type target: List[List[int]]
        :rtype: bool
        """
        n = len(mat)
        
        # Check exactly 4 orientations
        for _ in range(4):
            if mat == target:
                return True
            
            #  1: Transpose the matrix
            for i in range(n):
                for j in range(i + 1, n):
                    mat[i][j], mat[j][i] = mat[j][i], mat[i][j]
            
            #  2: Reverse every row 
            for i in range(n):
                mat[i].reverse()
                
        return False